opencensus/trace/ext/grpc/client_interceptor.py

```diff
@@ -0,0 +1,185 @@
+                                  grpc.UnaryStreamClientInterceptor,
+                                  grpc.StreamUnaryClientInterceptor,
+                                  grpc.StreamStreamClientInterceptor):
+
+    def __init__(self, tracer=None, host_port=None):
+        if tracer is None:
+            tracer = execution_context.get_opencensus_tracer()
+
+        self._tracer = tracer
+        self.host_port = host_port
+        self._propagator = binary_format.BinaryFormatPropagator()
+
+    def _start_client_span(self, method, grpc_type):
+        span = self._tracer.start_span(
+            name='[gRPC_client][{}]{}'.format(grpc_type, str(method)))
+
+        # Add the component grpc to span attribute
+        self._tracer.add_attribute_to_current_span(
+            attribute_key=attributes_helper.COMMON_ATTRIBUTES.get(
+                ATTRIBUTE_COMPONENT),
+            attribute_value='grpc')
+
+        # Add the host:port info to span attribute
+        self._tracer.add_attribute_to_current_span(
+            attribute_key=attributes_helper.GRPC_ATTRIBUTES.get(
+                GRPC_HOST_PORT),
+            attribute_value=self.host_port)
+
+        # Add the method to span attribute
+        self._tracer.add_attribute_to_current_span(
+            attribute_key=attributes_helper.GRPC_ATTRIBUTES.get(GRPC_METHOD),
+            attribute_value=str(method))
+
+        return span
+
+    def _end_span_between_context(self, current_span):
+        execution_context.set_current_span(current_span)
+        self._tracer.end_span()
+
+    def _intercept_call(
+            self, client_call_details, request_iterator, grpc_type):
+        metadata = ()
+        if client_call_details.metadata is not None:
+            metadata = client_call_details.metadata
+
+        # Start a span
+        current_span = self._start_client_span(
+            client_call_details.method,
+            grpc_type)
+
+        span_context = self._tracer.span_context
+        header = self._propagator.to_header(span_context)
+        grpc_trace_metadata = {
+            oc_grpc.GRPC_TRACE_KEY: header,
+        }
+        metadata = metadata + tuple(six.iteritems(grpc_trace_metadata))
+
+        client_call_details = _ClientCallDetails(
+            client_call_details.method,
+            client_call_details.timeout,
+            metadata,
+            client_call_details.credentials)
+
+        return client_call_details, request_iterator, current_span
+
+    def _callback(self, current_span):
+        def callback(future_response):
+            execution_context.set_current_span(current_span)
+            self._trace_future_exception(future_response)
+            self._tracer.end_span()
+
+        return callback
+
+    def _trace_future_exception(self, response):
+        # Trace the exception for a grpc.Future if any
+        exception = response.exception(timeout=TIMEOUT)
+
+        # If there is not exception, the attribute with null value will be
```

Comment:
I don't quite understand this comment.  Are you saying that exception might be None, or that str(exception) might be None?

Reply:
Thanks for noticing this, just added a check for whether the exception is None or not.